DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -266,56 +266,60 @@ class TrackerTopology {
   //these are clones of the old SiStripDetId
   uint32_t tobStereo(const DetId &id) const {
     if ( ((id.rawId() >>tobVals_.sterStartBit_ ) & tobVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_ );
+      return 1;
     } else { return 0; }
   }
 
   uint32_t tibStereo(const DetId &id) const {
     if ( ((id.rawId() >>tibVals_.sterStartBit_ ) & tibVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ );
+      return 1;
     } else { return 0; }
   }
 
   uint32_t tidStereo(const DetId &id) const {
     if ( ((id.rawId() >>tidVals_.sterStartBit_ ) & tidVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tidVals_.sterStartBit_) & tidVals_.sterMask_ );
+      return 1;
     } else { return 0; }
   }
 
   uint32_t tecStereo(const DetId &id) const {
     if ( ((id.rawId() >>tecVals_.sterStartBit_ ) & tecVals_.sterMask_ ) == 1 ) {
-      return ( (id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ );
+      return 1;
     } else { return 0; }
   }
 
   uint32_t tibGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ ) == 1 ) {
+    uint32_t testId = (id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_;
+    if ( testId == 1 ) {
       return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tibVals_.sterStartBit_) & tibVals_.sterMask_ ) == 2 ) {
+    } else if ( testId == 2 ) {
       return ( id.rawId() - 2 );
     } else { return 0; }
   }
 
   uint32_t tecGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ ) == 1 ) {
+    uint32_t testId = (id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_;
+    if ( testId == 1 ) {
       return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tecVals_.sterStartBit_) & tecVals_.sterMask_ ) == 2 ) {
+    } else if ( testId == 2 ) {
       return ( id.rawId() - 2 );
     } else { return 0; }
   }
 
   uint32_t tobGlued(const DetId &id) const {
-    if ( ((id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_ ) == 1 ) {
+    uint32_t testId = (id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_;
+    if ( testId == 1 ) {
       return ( id.rawId() - 1 );
-    } else if ( ((id.rawId()>>tobVals_.sterStartBit_) & tobVals_.sterMask_ ) == 2 ) {
+    } else if ( testId == 2 ) {
       return ( id.rawId() - 2 );
     } else { return 0; }
   }
```

Comment:
The idea is good, but I still need to test if testId is 1 or 2.
From the perspective of performanc it is not faster, though it is shorter.